proto/resources.proto

```diff
@@ -0,0 +1,65 @@
+syntax = "proto3";
+
+package rs.tokio.console.resources;
+
+import "google/protobuf/timestamp/timestamp.proto";
+import "google/protobuf/duration.proto";
+import "common.proto";
+
+// A task resource update.
+//
+// Each `ResourceUpdate` contains any resource data that has changed since the last
+// update. This includes:
+// - any new resources that were created since the last update
+// - the current stats for any resource whose stats changed since the last update
+message ResourceUpdate {
+    // A list of new resources that were created since the last `ResourceUpdate` was
+    // sent.
+    repeated Resource new_resources = 1;
+
+    // Any resource stats that have changed since the last update.
+    map<uint64, Stats> stats_update = 2;
+}
+
+// Static data recorded when a new resource is created.
+message Resource {
+    // The resources's ID.
+    //
+    // This uniquely identifies this resource across all *currently live*
+    // resources. This is also the primary way any operations on a resource
+    // are associated with it
+    common.SpanId id = 1;
+    // The numeric ID of the resources's `Metadata`.
+    common.MetaId metadata = 2;
+    // The resources's concrete rust type.
+    string concrete_type = 3;
+    // The kind of resource (e.g timer, mutex)
+    string kind = 4;
```

Comment:
We could save some allocations and wire size if we the `kind` a `oneof` with a bunch of "common" kinds, and another kind that is a string. Kinda like linkerd2's [`HttpMethod`](https://github.com/linkerd/linkerd2-proxy-api/blob/d9381f997306f5d68d262437f382789a891afc66/proto/http_types.proto#L7).

I don't mean this as a hard requirement _now_, but could be a good optimization maybe.

Reply:
yes, that makes sense indeeed